auth/src/VSCodeAzureSubscriptionProvider.ts

```diff
@@ -237,15 +242,21 @@ export class VSCodeAzureSubscriptionProvider extends vscode.Disposable implement
      *
      * @returns A client, the credential used by the client, and the authentication function
      */
-    private async getSubscriptionClient(tenantId?: string, scopes?: string[]): Promise<{ client: SubscriptionClient, credential: TokenCredential, authentication: AzureAuthentication }> {
+    private async getSubscriptionClient(tenantId?: string, scopes?: string[], session?: vscode.AuthenticationSession, account?: vscode.AuthenticationSessionAccountInformation): Promise<{ client: SubscriptionClient, credential: TokenCredential, authentication: AzureAuthentication }> {
         const armSubs = await import('@azure/arm-resources-subscriptions');
-        const session = await getSessionFromVSCode(scopes, tenantId, { createIfNone: false, silent: true });
         if (!session) {
-            throw new NotSignedInError();
+            if (account) {
+                session = await getSessionFromVSCode(scopes, tenantId, { createIfNone: false, silent: true, account });
+            } else {
+                session = await getSessionFromVSCode(scopes, tenantId, { createIfNone: false, silent: true });
+            }

```

Comment:
I believe this would be the same, but without the if statement
```suggestion
            session = await getSessionFromVSCode(scopes, tenantId, { createIfNone: false, silent: true, account });
```

Reply:
![image](https://github.com/user-attachments/assets/f501c3ce-73e1-4537-af83-5b0c4ba64437)